core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalWriteConverterCtx.scala

```diff
@@ -55,6 +55,7 @@ class ExternalWriteConverterCtx(nodeDesc: NodeDesc, totalNumOfRows: Int) extends
   override def put(colIdx: Int, data: Float) = externalFrameWriter.sendFloat(data)
   override def put(colIdx: Int, data: Double) = externalFrameWriter.sendDouble(data)
   override def put(colIdx: Int, data: java.sql.Timestamp) = externalFrameWriter.sendTimestamp(data)
+  override def put(colIdx: Int, data: java.sql.Date) = externalFrameWriter.sendTimestamp(new java.sql.Timestamp(data.getTime))
```

Comment:
To get rid of the allocation we could do 
`
override def put(colIdx: Int, data: java.sql.Date) = externalFrameWriter.sendLong(data.getTime)
` 
because in writer context on h2o side we handle timestamp in the same way as the rest of numbers